app/services/IdentityService.scala

```diff
@@ -56,10 +42,37 @@ class IdentityService(identityApiClient: IdentityApiClient) extends LazyLogging
       }
     }
   }
+
+  def updateUserDetails(personalData: PersonalData, userId: UserId): Future[Unit] = 
+    identityApiClient.updateUserDetails(
+      JsObject(Map("privateFields" -> personalData.convertToPrivateFields())), 
+      userId.id)
+    .map(_ => Unit)
+
 }
 
 object IdentityService extends IdentityService(IdentityApiClient) {
   class IdentityGuestPasswordError(jsonMsg: String) extends RuntimeException(s"Cannot set password for Identity guest user. Json response form service: $jsonMsg")
+
+  implicit class PersonalDataJsonSerialiser(personalData: PersonalData) {
+    implicit def convertToPrivateFields(): JsObject = JsObject(Map(
```

Comment:
...still you can probably lose the implicit keyword next to each method definition